src/rgw/rgw_sal_motr.cc

```diff
@@ -1297,26 +1297,13 @@ int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx,
   if (this->category == RGWObjCategory::MultiMeta)
     return 0;
 
-  string bname, key;
-  if (target_obj) {
-    bname = get_bucket_name(target_obj->bucket.tenant, target_obj->bucket.name);
-    key   = target_obj->key.to_str();
-  } else {
-    bname = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
-    key   = this->get_key().to_str();
-  }
-  ldpp_dout(dpp, 20) <<__func__<< ": " << bname << "/" << key << dendl;
-
   // Get object's metadata (those stored in rgw_bucket_dir_entry).
   bufferlist bl, update_bl;
-  string bucket_index_iname = "motr.rgw.bucket.index." + bname;
-  if (this->store->get_obj_meta_cache()->get(dpp, key, bl)) {
-    // Cache miss !
-    int rc = this->store->do_idx_op_by_name(bucket_index_iname, M0_IC_GET, key, bl);
-    if (rc < 0) {
-      ldpp_dout(dpp, 0) <<__func__<< ": Failed to get object's entry from bucket index. rc=" << rc << dendl;
-      return rc;
-    }
+  string bname, key;
+  int r = read_obj_attrs(dpp, bname, key, bl, target_obj);
+  ldpp_dout(dpp, 20) <<__func__<< ": " << bname << "/" << key << dendl;
```

Comment:
this log statement giving bucket name & key should be after 'r' check. same for line 1343